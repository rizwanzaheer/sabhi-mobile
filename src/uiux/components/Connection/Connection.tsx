import * as React from 'react'
import { TouchableOpacity } from 'react-native'
import Container from '../Container/Container'
import Text from '../Text/Text'
import Avatar from '../Avatar/Avatar'
import { withTheme } from '../../theming'

interface ConnectionProps {
  onPress: () => void
  did: string
  isManaged: boolean
  shortId: string
  profileImage?: string
  theme: any
}

const Connection: React.FC<ConnectionProps> = ({ onPress, did, isManaged, shortId, profileImage, theme }) => {
  const displayDid = shortId.startsWith('did:ethr:') ? shortId.slice(9, -4) : shortId
  const source = profileImage ? { source: { uri: profileImage } } : {}
  const gravatar: any = shortId.startsWith('did:ethr:') ? { gravatarType: 'retro' } : {}

  return (
    <TouchableOpacity onPress={onPress}>
      <Container alignItems={'center'} justifyContent={'center'} padding={10}>
        <Container marginBottom={10}>
          {isManaged && (
            <Container
              alignItems={'center'}
              justifyContent={'center'}
              backgroundColor={theme.colors.theme.CONFIRM}
              viewStyle={{
                borderWidth: 1,
                borderColor: '#ffffff',
                width: 14,
                height: 14,
                borderRadius: 10,
                position: 'absolute',
                zIndex: 100,
                bottom: 0,
                right: 3,
              }}
            ></Container>
          )}
          <Avatar size={60} address={did} title={shortId} {...source} {...gravatar} />
        </Container>
        <Text textStyle={{ fontSize: 14 }}>{displayDid}</Text>
      </Container>
    </TouchableOpacity>
  )
}

export default withTheme(Connection)
