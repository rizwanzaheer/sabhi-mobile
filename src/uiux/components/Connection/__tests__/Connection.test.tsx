import * as React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import Connection from '../Connection'

// onPress: () => void
// did: string
// isManaged: boolean
// shortId: string
// profileImage?: string
// theme: any

describe('Component(snapshot): Connection', () => {
  it('should render with basic shortId', () => {
    const onPress = jest.fn()
    const tree = render(
      <Connection
        onPress={onPress}
        did={'did:ethr:0x1fcc123ee34812db4bec937d7acdc7ca61d990a6'}
        isManaged={false}
        shortId={'did:ethr:0x1fcc...90a6'}
      />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render with custom shortId and profileImage', () => {
    const onPress = jest.fn()
    const tree = render(
      <Connection
        onPress={onPress}
        did={'did:ethr:0x1fcc123ee34812db4bec937d7acdc7ca61d990a6'}
        isManaged={true}
        shortId={'Sarah'}
        profileImage={'http://'}
      />,
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('should call onPress when tapped', () => {
    const onPress = jest.fn()
    const { getByText } = render(
      <Connection
        onPress={onPress}
        did={'did:ethr:0x1fcc123ee34812db4bec937d7acdc7ca61d990a6'}
        isManaged={true}
        shortId={'Sarah'}
        profileImage={'http://'}
      />,
    )

    fireEvent.press(getByText(/Sarah/i))
    expect(onPress).toHaveBeenCalled()
  })
})
