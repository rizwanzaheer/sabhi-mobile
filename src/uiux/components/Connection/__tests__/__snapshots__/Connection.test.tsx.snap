// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshot): Connection should render with basic shortId 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": "center",
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": 10,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": 10,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#0078ff",
            "borderRadius": 30,
            "flex": undefined,
            "flexDirection": undefined,
            "height": 60,
            "justifyContent": "center",
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": 60,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#ffffff",
              "flexWrap": "wrap",
              "fontSize": 30,
              "fontWeight": "bold",
            }
          }
        >
          Di
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 14,
        }
      }
    >
      0x1fcc...
    </Text>
  </View>
</View>
`;

exports[`Component(snapshot): Connection should render with custom shortId and profileImage 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": "center",
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": 10,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": 10,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3DCF8A",
            "borderColor": "#ffffff",
            "borderRadius": 10,
            "borderWidth": 1,
            "bottom": 0,
            "flex": undefined,
            "flexDirection": undefined,
            "height": 14,
            "justifyContent": "center",
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "position": "absolute",
            "right": 3,
            "width": 14,
            "zIndex": 100,
          }
        }
      />
      <Image
        source={
          Object {
            "uri": "http://",
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "#0078ff",
              "borderRadius": 30,
              "height": 60,
              "width": 60,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      />
    </View>
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 14,
        }
      }
    >
      Sarah
    </Text>
  </View>
</View>
`;
